### PR TITLE
Loading to Viewer

### PR DESCRIPTION
**Pull request by haitheredavid**

### CODE REVIEW

File: viewto-frontend/src/components/Viewer/Viewer.tsx [hunk 1]
```diff
@@ -1,8 +1,67 @@
...
+                const referenceObj = versionRef.reference;
+
+                const url = `https://sasaki.speckle.xyz/streams/${scenario.project.id}`;
+                const objUrl = `${url}/objects/${referenceObj}`;
+
+                console.log(`getting object ${objUrl}`);
+
+                await viewer.current?.loadObjectAsync(objUrl, import.meta.env.VITE_SPECKLE_TOKEN);
+            }
+        })();
```
Comment (arminakvn) [inline viewto-frontend/src/components/Viewer/Viewer.tsx:64]: @haitheredavid  @ericyoungberg I wonder if eventually we want to do a cleanup at the end of the useEffect hook ([david if you see this](https://react.dev/reference/react/useEffect#useeffect)) to unload the objects from the viewer?

**Review by arminakvn — APPROVED** [review]

File: viewto-frontend/src/components/Viewer/Viewer.tsx (hunk 1 above)
Comment (haitheredavid) [reply viewto-frontend/src/components/Viewer/Viewer.tsx:64]: @arminakvn that's good to note! Is this with the intention that we need to unload all of the objects when we destroy the viewer ? I think the only time we have to unload the objects is if we change the study loaded from speckle

Comment (ericyoungberg) [reply viewto-frontend/src/components/Viewer/Viewer.tsx:64]: @arminakvn perhaps! The objects are loaded into the viewer which is scoped to the component, so when the component dismounts, it should cleanup the viewer. We usually want to unload from things loaded in via effect if they are loaded into something whose lifecycle is not scoped to the component. Is there any indication of a memory leak?

File: viewto-frontend/src/models/ReferenceObject.ts [hunk 2]
```diff
@@ -33,17 +31,26 @@ export default class ReferenceObject {
...
+    }
+
     @action
     async load() {
         if (!this.isLoading) {
             const { speckle, project } = (stores as Stores).scenario;
             this.setIsLoading();
             const obj = await speckle.Project(project.id).Version(this.id).get;
             this.setData(obj);
+            this.setReference(obj.referencedObject);
```
Comment (ericyoungberg) [inline viewto-frontend/src/models/ReferenceObject.ts]: This is duplicate state. We should either use this value from the `data` property set on line 52 or create a computed property for this value that points to `data.referencedObject`.

**Review by ericyoungberg — CHANGES_REQUESTED** [review]

File: viewto-frontend/src/models/ReferenceObject.ts (hunk 2 above)
Comment (haitheredavid) [reply viewto-frontend/src/models/ReferenceObject.ts]: made these changes to computed

**Review by ericyoungberg — APPROVED** [review]

**Pull request merged**